### ISSUE TITLE
Please enter a valid length!

### ISSUE THREAD

**Issue by lightnb** [phpmyadmin/phpmyadmin#13158]
### Steps to reproduce
1. Choose to create a new table
2. Name the fields, select types, add indexes and options
3. Attempt to save.

### Expected behavior
Since nothing is invalid, it should save and create the new table.

### Actual behavior
Many times, it does save the new table, but for some reason which I cannot determine, there is a 1 in 25 chance (approximately) of an error message popping up in a javascript alert which reads "Please enter a valid length!". Nothing is actually wrong.

If you start over by choosing to create a new table and re-enter all of the exact same data, it works fine. I suspect there may be some issue with the order in which fields are filled in.

I have attempted to re-produce this and it does recur often, but the best I can come up with so far is "get unlucky".

### Server configuration
**Operating system**: Ubuntu 16.04

**Web server:** NginX 1.10.0

**Database:** MySQL

**PHP version:** 7.0.13

**phpMyAdmin version:**  4.5.4.1deb2ubuntu2

### Client configuration
**Browser:** Firefox 45 LTS

**Operating system:** Mint 18

**Comment by OlafvdSpek** [comment]
> phpMyAdmin version: 4.5.4.1deb2ubuntu2

I think you should try 4.7 first..

**Comment by madhuracj** [comment]
Try 4.7 on our demo server (https://demo.phpmyadmin.net/QA_4_7, username: root, no password)
If you can recreate the issue there, please provide us with a screenshot of the values you've entered, so it would be easy for us to recreate and fix the issue.

**Comment by poush** [comment]
I also encountered this error once in one of my competitions where we were using PHPMyAdmin for database importing. I wasn't able to check the version due to short of time there. Although that PHPMyAdmin was part of Xampp installation on that system. The problem was occurring continuously until I reloaded PHPMyAdmin.
I was stuck with importing process for 10 minutes due to this PHPMyAdmin issue that day.
@lightnb Please try this on 4.7 and see if you can recreate it.

**Comment by lightnb** [comment]
I can't even recreate it on the
      affected version. It *reoccurs* frequently, but there is no
      consistent pattern to when it occurs. If it happens, and then I
      reload the add columns screen and enter the *exact same data
      again*, it *might* work. Thus, even with a screen-shot of the
      exact data, there is no guarantee that the error will occur with
      that exact same data a second time. The error is caused by
      something other than the data entered, possibly a race condition
      or a validation logic error depending on the order in which fields
      are filled out? It may have something to do with the index pop up
      boxes. (By the way, I wish there was an option to disable the
      index pop ups and just use single indexes on everything by default
      without asking).
      
      
      On 04/08/2017 07:09 AM, Piyush Agrawal wrote:
    
    
      I also encountered this error once in one of my competitions
        where we were using PHPMyAdmin for database importing. I wasn't
        able to check the version due to short of time there. Although
        that PHPMyAdmin was part of Xampp installation on that system.
        The problem was occurring continuously until I reloaded
        PHPMyAdmin.
        I was stuck with importing process for 10 minutes due to this
        PHPMyAdmin issue that day.
        @lightnb Please try this on 4.7 and
        see if you can recreate it.
      —
        You are receiving this because you were mentioned.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/phpmyadmin/phpmyadmin","title":"phpmyadmin/phpmyadmin","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/phpmyadmin/phpmyadmin"}},"updates":{"snippets":[{"icon":"PERSON","message":"@poush in #13158: I also encountered this error once in one of my competitions where we were using PHPMyAdmin for database importing. I wasn't able to check the version due to short of time there. Although that PHPMyAdmin was part of Xampp installation on that system. The problem was occurring continuously until I reloaded PHPMyAdmin.\r\nI was stuck with importing process for 10 minutes due to this PHPMyAdmin issue that day.\r\n@lightnb Please try this on 4.7 and see if you can recreate it. "}],"action":{"name":"View Issue","url":"https://github.com/phpmyadmin/phpmyadmin/issues/13158#issuecomment-292711170"}}}

**Comment by udan11** [comment]
The only snippet of code that may cause that problem is found [here](https://github.com/phpmyadmin/phpmyadmin/blob/master/js/functions.js#L835) and looks correct.

`field_X_1` is the "Name" field and `field_X_3` is the "Length/Values" field. It checks if there is a valid number and the field name is not empty, which is correctly.

**Comment by nijel** [comment]
Anybody can reproduce this or current version? If not, it's probably close this report as it has been fixed meanwhile...

**Comment by aornt** [comment]
I see this thread is closed, but in version 4.7.4 (up to date) the problem still is there.

**Comment by OlafvdSpek** [comment]
Could you provide instructions on how to reproduce this, preferably via https://www.phpmyadmin.net/try/ ?

**Comment by aornt** [comment]
At this moment I have logged off and on in phpMyAdmin and the problem has solved for the moment.
It's just very annoying that having filled in all 15 field definitions you have to redo it.
I filled now just 2 rows and then already created the database. This worked OK.
(before the log-on log-off it did also not work with 2 rows)
Maybe the error appeared because I used the "add column"-button at the top, in order to extend the input rows for adding more fields.

I'll try to determinate the conditions more precise when it happens again.

**Comment by nijel** [comment]
Thanks, having reproducer would certainly allow us to fix this.

**Comment by PaapaAsare** [comment]
SOLVED
if you have set the datatype of any of the fields to "VARCHAR" then you would have to set the length of those fields.  under LENGTH/VALUES, put any number there.

**Comment by lightnb** [comment]
That's actually not the solution- this
      is not operator error. The issue is that sometimes it thinks the
      length is empty when you have in fact filled it in. The exact
      steps to trigger this broken condition are not known, but it does
      reoccur with some frequency, and the only way to fix it is to
      reload the form and start over once it gets stuck in this mode.
      
      
      On 02/09/2018 03:05 AM, PaapaAsare wrote:
    
    
      SOLVED
        if you have set the datatype of any of the fields to "VARCHAR"
        then you would have to set the length of those fields. under
        LENGTH/VALUES, put any number there.
      —
        You are receiving this because you were mentioned.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/phpmyadmin/phpmyadmin","title":"phpmyadmin/phpmyadmin","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/phpmyadmin/phpmyadmin"}},"updates":{"snippets":[{"icon":"PERSON","message":"@PaapaAsare in #13158: SOLVED\r\nif you have set the datatype of any of the fields to \"VARCHAR\" then you would have to set the length of those fields.  under LENGTH/VALUES, put any number there. "}],"action":{"name":"View Issue","url":"https://github.com/phpmyadmin/phpmyadmin/issues/13158#issuecomment-364363252"}}}

**Comment by pagan-bear** [comment]
I agree with @lightnb above.  I have just gotten this error now.  I always enter a length for varchar fields.  What ever "happens", until you do a browser page reload the error seems to repeat.  One thing I did notice is that after the error occurs you can subsequently, successfully add table rows one-at-a-time.  After the initial exception I tried to run SQL manually to add table rows, but this failed with the above exception too.

![phpmyadmin - version details](https://user-images.githubusercontent.com/33454013/36071905-7de08b3e-0f16-11e8-85a6-37d8bca94b8f.png)

**Comment by emmjk** [comment]
I also face the same problem .

**Comment by alea123** [comment]
today i got this error the first time on:

Datenbank-Server
    Server: 127.0.0.1 via TCP/IP
    Server-Typ: MySQL
    Server-Version: 5.7.24-1 - (Debian)
    Protokoll-Version: 10
    Benutzer: root@127.0.0.1
    Server-Zeichensatz: UTF-8 Unicode (utf8)

phpMyAdmin
    Versionsinformationen: 4.3.9

i know its not the latest version of all, but never had this error before

**Comment by williamdes** [comment]
Seems that some people still have this issue.
Demo servers (root, no password)
- [`5.0.0-dev`](https://demo.phpmyadmin.net/master-config/)
- [`4.8.x`](https://demo.phpmyadmin.net/QA_4_8/)
TLDR; If someone can try to reproduce on demo-servers ?

**Comment by williamdes** [comment]
Still no feedback, closing.
Please try on latest version and send detailed steps to reproduce, and we will fix the issue !

**Comment by official-garlic** [comment]
The problem is still present in the built as of today. I'm sorry I am not able to reproduce the error because I don't know how to use and connect to the demo version.

**Comment by OlafvdSpek** [comment]
https://demo.phpmyadmin.net/master-config/

**Comment by VanshikaSingolia** [comment]
Try giving a value for length in all attributes having VARCHAR data type,it actually works...!!
